server/login.js

```diff
@@ -0,0 +1,77 @@
+import { Router } from "express";
+import query from "./db.js";
+import jwt from "jsonwebtoken";
+import bcrypt from "bcrypt";
+import { authenticateToken } from "./utils/authMiddleware.js";
+
+const router = Router();
+
+// router.post("/login", async (req, res) => {
+//   try {
+//     const { email, password } = req.body;
+
+//     // Check if the user exists
+//     const result = await query.query(
+//       "SELECT * FROM users WHERE email = $1",
+//       [email]
+//     );
+
+//     const user = result.rows[0];
+
+//     if (!user) {
+//       return res.status(401).json({ error: "Invalid credentials" });
+//     }
+
+//     // Validate password
+//     const isValidPassword = await bcrypt.compare(password, user.password);
+
+//     if (!isValidPassword) {
+//       return res.status(401).json({ error: "Invalid credentials" });
+//     }
+
+//     // Generate JWT token
+//     const token = jwt.sign({ userId: user.id }, process.env.JWT_SECRET, {
+//       expiresIn: "1h",
+//     });
+
+//     res.json({ message: "Login successful", token });
+//   } catch (error) {
+//     console.error(error);
+//     res.status(500).json({ error: "Server error" });
+//   }
+// });
+
+
+router.post("/", async (req, res) => {
+  try {
+    const { email, password } = req.body;
+
+    // Hardcoded test credentials
+    const testEmail = "test@example.com";
+    const testPassword = "password123";
```

Comment:
looks good but since you handled student registration, you can fetch the student and compare the password with the one stored in the database